include/seqan3/argument_parser/validators.hpp

```diff
@@ -525,6 +565,113 @@ struct default_validator
     }
 };
 
+/*!\brief A helper struct to chain validators recursively via the pipe operator.
+ *\ingroup argument_parser
+ *\implements seqan3::validator_concept
+ *
+ *\details
+ *
+ * Note that both validators must operate on the same value_type in order to
+ * avoid unexpected behaviour and ensure that the seqan3::argument_parser::add_option
+ * call is well-formed. (add_option(val, ...., validator) requires
+ * that val is of same type as validator::value_type).
+ */
+template <validator_concept validator1_type, validator_concept validator2_type>
+//!\cond
+    requires std::Same<typename validator1_type::value_type, typename validator2_type::value_type>
+//!\endcond
+class validator_chain_adaptor
+{
+public:
+    //!\brief The underlying type in both validators.
+    using value_type = typename validator1_type::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief The default constructor is explicitly deleted.
+    validator_chain_adaptor() = delete;
```

Comment:
Just a general question. You explicitly delete the default construction here. And the general design is that it is not used somewhere else than for the  argumentparser. But are all your validators not default constructible?
So I would propose to allow validator to be a semi-regular type and in case of default construction of the adaptor use the default_validator which always passes.

Reply:
If we really want to allow default constructibility, I would propose to specifically create a validator that in the context allows every value (type dependent of course).
e.g. the regex_validator would be defaulted to `val{".*"}`